Adds modular pc vendors

Wilchenx opened the pull request:
## About The Pull Request
Adds a new modular pc vendor that can be found generally near the science departament lobby

## Why It's Good For The Game
Right now modular pcs, even at their most basic state are locked away to only one departament that we all hate and love taking things away from, Science.

So its time for them to share more of their deluxe silicate toys.

This is also a moneysink (its almost like a """""economy""""" where money just keeps flowing in and there are no sinks for it is poorly designed)
## Changelog
:cl:
add: The modular pc vendor
del: Random plant on deltastation (his name was rodger)
del: A table with pamphlet on metastation
del: A table with wires and circuitry from pubbystation
del: Crucial emptyspace from boxstation, further cluttering the space around insulated gloves vendor
/:cl:

Commit:
Merge pull request #3 from tgstation/master

Update

Commit:
monkeysink

Commit:
yes

Commit:
Deluxe

Commit:
Update modularpc.dm

yoyobatty committed:
Fix blastdoors/shutters button spam (#43577)

yoyobatty committed:
Fix being able to build multiple terminals for SMES's (#43579)

Tlaltecuhtli commented on code/modules/vending/modularpc.dm:
```diff
@@ -0,0 +1,29 @@
+	products = list(/obj/item/modular_computer/laptop = 4,
+					/obj/item/modular_computer/tablet = 4,
+					/obj/item/computer_hardware/hard_drive = 4,
+					/obj/item/computer_hardware/hard_drive/small = 4,
```
why have the downgrade at the same price and with the same availability of the better one

JStheguy replied:
Well, regular-size hard drives won't fit into tablets, iirc.
Though it is still kind of silly, regardless. @Wilchenx if you want to change this, just set the custom_price var of either the small or normal drive to something more suitable.

Wilchenx replied:
god i love undocumented features
![image](https://user-images.githubusercontent.com/25509881/56046436-865dd880-5d43-11e9-83ba-eb25fb1dbb2f.png)

optimumtact replied:
they say as they stare at the documentation

optimumtact replied:
can you confirm you've fixed this

Wilchenx replied:
I've yet not changed the price, there's also still a merge conflict on donutstation

MMMiracles committed:
adds 2 million lightswitches to donutstation (#43377)

* my l key broke while making this it was very annoying

* removes rogue pixel_x/y 0's

* shizbot

* too many intercomms ruins the station

ArcaneMusic commented:
Solid hustle, maybe this will convince people to add more functionality to tablets/modular computers.

pireamaineach committed:
Ports far too many haircuts and beards. (#43564)

* FuCK

* Adds far too many haircuts.

Help.

* Update sprite_accessories.dm

* Update human_face.dmi

* Update human_face.dmi

* Update human_face.dmi

Tlaltecuhtli committed:
[ready]secway + atv nerfs (#43396)

* 1

* dont commit this

* commit this

* commit erefasdagsdbsvrsccsdsdfsdfs

* changes

* fdf

* fixes bullet_act

* obj_break

* (____Y____)

SpaceManiac committed:
Fix mapmerge misreading maps with two columns per block

MMMiracles committed:
Minor Donutstation tweaks (#43609)

Removes redundant tech fab/lathe from atmos, as well as the pipe
dispensers. put 2 rad suit lockers and a regular fire locker in their
place. CE's windows are electrified.

RaveRadbury committed:
pAIs eject into hands (#43592)

Currently pAIs cannot be configured for DNA and secondary directives
while inside a PDA. If I need to eject my pAI from my PDA, why would I
want to drop it on the floor, especially when it hasn't been configured?

An activated pAI is too valuable to drop from your PDA directly to the
ground.

81Denton committed:
Standardize cable colors across maps (#43599)

This PR standardizes cable colors: Power source to SMES is red, SMES to
main grid is yellow. Cables that can connect the main grid to AI
sat/telecomms sat are cyan.

It also fixes some odd wiring at the Pubby incinerator, the zoo ruin and
the BMP ship ruin.

This would probably be a good testmerge candidate, since it touches a lot
of map code. Aside from the three files where I tweaked/fixed cables, the
only effect should be different colors.

Cable colors are all over the place, which causes a lot of confusion for
players that haven't memorized every single area's cable layout. Cables
are probably the smallest and most annoying objects to interact with
ingame, so what they're doing should be obvious at first glance.

I chose red for power source -> SMES wires, since these are most likely
to really hurt when you touch them. Yellow is IMO the best color for the
main grid, since it doesn't overlap with scrubber pipes like red cables
do.

AnturK committed:
Buff beetlejuice component (#43581)

Forgot to add this when I changed it to regex.

WJohn committed:
Fix movement hunger costs (#43607)

It seems it was intended to make movement cost a bit of hunger per tile,
but was actually only happening while you were floating (despite the
comment saying it should be easy).

This reverses that so it's now working as intended.

Testing shows this increases nutrition by about 5.6 per minute of
non-stop (non-floating) running movement. Walking cuts this cost by more
than half as not only do you cover fewer tiles in a minute, but walking
also costs half as much in tile-tile movement.

If you're on meth, expect to get quite hungry from zooming around the
station like a madman at 300 mph, which I think makes quite a bit of
sense.

RaveRadbury committed:
pAI cards update on availability while in inventory (#43587)

pAI cards will now provide updates about available personalities even
when in a player's inventory. Improves the pAI matchmaking system, making
personalities more likely to be noticed and downloaded.

terranaut1 committed:
Add salt to ramen noodles (#43586)

Throw it at salty greytiders you toolboxed. Also useful for ghettochem.

imsxz committed:
[s] adds actual logging to spray bottles (#43627)

adds logging to spraybottle usage. spraying in general is logged in game to track people spreading plasma/lube/whatever. theres transferring reagents to people via spraying logged in attack.

JJRcop committed:
[READY] Adds selector lists to SDQL (#43383)

* [SDQL2] Adds selector arrays

* SDQL2 documentation spec updated slightly

* SDQL2 selector array docs + tweaks

If a single list is the only result of a MAP statement,
meaning objs.len == 1, AND it's a list, it will unwrap
the result from its containing list and return it directly.

The selector array logic keeps track of the state of the query
when it starts the first Search. It won't change the state back
to Execute until the top-level is reached, which sets a var
to keep track.

* SDQL2 MAP change from objs.len to length(objs)

* [SDQL2] Reverts MAP list unwrap

* [SDQL] Move (expr) from expr, unary_expr to var

This should result in the same functionality but instead of expressions and unary_expressions processing sub expressions, variables do
This means that expression->value->variable is now completely recursable

* [SDQL2] Update selector list documentation

SpaceManiac committed:
Fix unit confusion in radiation collector array description (#43523)

SpaceManiac committed:
Include BYOND build when using Report Issue button (#43545)

RaveRadbury committed:
Chefs can shower with their mustache on (#43614)

* Update miscellaneous.dm

* Update miscellaneous.dm

Ordonis committed:
Adds Several New Alcoholic Beverages (#43416)

* Adds Champagne, Bug Spray, and Wizz Fizz

Three new drinks. How did we not have champagne already?

* REEEE COMMA

REEEEE

* Fixing stupid mistake and beginning BUG fixing

Observe good pun

* Less direct referencing of vars and defines

Also screaming

* Adds applejack, fixes small errors in coloration

Horsepuns HONK

* Adds sprites, includes sprites for bottles of champagne/applejack

Does what it says on the tin.

* Adds sprites.

* Fixes champagne and applejack being nice drinks.

* Adds bottles for the new pure drinks

Does not add them to the boozeomatic yet.

* Switches to Nemvar/Kryson sprites

These are better. Also it means wizz fizz is animated.

* nerfs bugspray

* Nerfs Wizzfizz and Bugspray, adds Jack Rose

see title

* Changes how wiz is checked, removes vestigal code, FUCK

* Changes screaming to on_mob_add like ATATH suggested earlier and cleans up healing code

*scream

* Adds applejack and Champagne to Booze-o-mat

Commit:
Changes the for-flavour One Mind objective into a functional Protect objective (#43624)

* Update hivemind.dm

* yikes

SpaceManiac committed:
Add a missing period when splitting stacks (#43604)

anconfuzedrock committed:
makes gunz black (#43576)

81Denton committed:
Adds a few more golem names (#43573)

* Adds a few more golem names

* Update golems.dm

* Removes overridden bananium golem special names, adds "Bottle"

Fikou committed:
makes the medical hardsuit faster (#43597)

* Update hardsuit.dm

* Update hardsuit.dm

AnturK committed:
Make RCD not work on reinforced doors (#43598)

This is not really a big change since there's almost certainly walls around
the door.

RaveRadbury committed:
Synthetizer is now Synthesizer (#43635)

I'm doing a lot of work on the pAI stuff right now and mad props to
whoever set the loudness booster up, but this typo is killing me every
time I see it.

blessedmulligan committed:
Adjuste text for viruses causing eyeballs to detach (#43629)

The "Hyphema" virus symptom now makes eyeballs fall "out of" their sockets
and not "off" them.

Time-Green committed:
Fixes some inconsistencies in genetics (#43583)

* Adds chromosome interactions with more mutations

* Makes upgrades affect printing cooldown

* Fixes a runtime with monkey dna and human eyes

RaveRadbury committed:
Expands service channel access to HoP-reporting jobs. (#43601)

* Gave service headsets to jobs that report to HoP (not assistants, obviously). Created a headset and encryptionkey for lawyers that is sec service (similar to genetics)

* Typo

* Less copypasta more sensemaking

sohpeach committed:
TGUI buttons do not fully grey out while the UI is still updating (#43540)

* desaturate TGUI buttons instead of disabling when status is UI_UPDATE

* TGUI buttons changed to stripped-out style and never fully grey out

imsxz committed:
adds overdose to syndie medborg nanites (#43541)

* Automatic changelog generation for PR #42911 [ci skip]

* x

* nonlethal

* copypasta strikes

sohpeach committed:
Ghosts can analyze gas mixes and some new objects can be analyzed (#43502)

* Ghosts can analyze gas mixes and some new objects can be analyzed

* Replace show_air_status_to with atmosanalyzer_scan

JJRcop committed:
Remove infectability from xeno nest lavaland ruin (#43616)

* Remove live eggs from xeno nest lavaland ruin

* Remove dead facehuggers from xeno lavaland ruin

actioninja committed:
Ports Hippie's progress bar animations (#43492)

* progress bar anims

* gold final stage

* technical improvements

* special failure animation + code improvement

* mm grayons

* better animation

* less magic number + garbage collection safety tweak

* I forgot to actually include the garbage collection tweak

* even more garbage collection safety

optimumtact committed:
Merge pull request #43628 from AutomaticFrenzy/patch/minskystation

Fix mapmerge misreading maps with two columns per block

actioninja committed:
Converts almost every single sound to mono 44.1khz, recuts some sounds. (#43550)

About The Pull Request

As title, except a clarification: sounds that play with no source or direction are still stereo due to that the stereo will actually do something and not just be wasted space.

On a technical level, what I actually did was reexport (nearly) every single sound effect to mono 44.1khz sample rate Ogg at quality preset 5. This resulted in some small increases in file size in some places but other than that reduced file size across the board.

A couple of sounds that were wavs for whatever reason were also converted to ogg, this resulted in some code changes to change the referenced sound effect.

Some sounds were also recut. The grille hit and punches in particular are common sounds that had large and noticeable gaps at the beginning.

All of this has resulted in the compiled rsc file going from 102.7 mb to 83.2 mb.
Why It's Good For The Game

Free space with no (effective) drop in quality.
A standard for audio going forward. (that will be inevitably ignored)
Tighter sounds in some places.
Changelog

cl
soundadd: Downmixed every sound to mono. This shouldn't have resulted in any changes in audio, byond downmixes anyways.
soundadd: Recut some sounds. Sounds shouldn't have large silent/noise gaps at the start anymore
/cl

actioninja committed:
riot armor buff (#43590)

Commit:
Killing the One Mind's host destroys the One Mind (#43623)

* Update hivemind.dm

* Update hivemind.dm

* Update code/modules/antagonists/hivemind/hivemind.dm

Co-Authored-By: Kierany9 <Kierany9@users.noreply.github.com>

Commit:
n

Commit:
c

Commit:
Update Donutstation.dmm

Commit:
Update Donutstation.dmm

Commit:
Merge branch 'moneysink' of https://github.com/Wilchenx/tgstation into moneysink

Wilchenx commented:
i claim the fifth
